src/services/quandl-service.js

```diff
@@ -45,6 +48,10 @@
     }
 
     function processResponse(json) {
+        if (!json || json.quandl_error) {
+            return;
+        }
+
```

Comment:
Both the call to `filterByDate` and `processResponse` are preceded by a line looking like `json = angular.fromJson(data)`. I think it might make more sense (and reduce duplication!) to move the sanitising checks above into a new method wrapping the `fromJson` call, instead of having them in both these methods.